linter/pwnedkeys/handler.go

```diff
@@ -0,0 +1,113 @@
+package pwnedkeys
+
+import (
+	"context"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+
+	"github.com/pkimetal/pkimetal/config"
+	"github.com/pkimetal/pkimetal/linter"
+)
+
+type Pwnedkeys struct{}
+
+var httpClient http.Client
+
+func init() {
+	httpClient = http.Client{
+		Timeout: config.Config.Linter.Pwnedkeys.HTTPTimeout,
+	}
+
+	// Register pwnedkeys.
+	(&linter.Linter{
+		Name:         "pwnedkeys",
+		Version:      "1.0.0",
+		Url:          "https://pwnedkeys.com/api/v1",
+		Unsupported:  linter.NonCertificateProfileIDs,
+		NumInstances: config.Config.Linter.Pwnedkeys.NumGoroutines,
+		Interface:    func() linter.LinterInterface { return &Pwnedkeys{} },
+	}).Register()
+}
+
+func (l *Pwnedkeys) StartInstance() (useHandleRequest bool, directory, cmd string, args []string) {
+	return true, "", "", nil // The pwnedkeys API is called from Goroutine(s) in the pkimetal process, so there are no "external" instances.
+}
+
+func (l *Pwnedkeys) StopInstance(lin *linter.LinterInstance) {
+}
+
+func (l *Pwnedkeys) HandleRequest(lin *linter.LinterInstance, lreq *linter.LintingRequest, ctx context.Context) []linter.LintingResult {
+	var lres []linter.LintingResult
+	var httpRequest *http.Request
+	var err error
+	s := sha256.Sum256(lreq.Cert.RawSubjectPublicKeyInfo)
+	if httpRequest, err = http.NewRequest(http.MethodGet, fmt.Sprintf("https://v1.pwnedkeys.com/%s", hex.EncodeToString(s[:])), nil); err != nil {
```

Comment:
If you're accepting a context.Context, you might as well do NewRequestWithContext(ctx, ...) and pass it along

Reply:
Thanks.  Addressed by https://github.com/pkimetal/pkimetal/pull/183/commits/e36cf871ad33153db43bd5339c8a7b1977aaf8f5